components/encryption/src/master_key/kms.rs

```diff
@@ -57,21 +58,24 @@ impl KmsBackend {
 
         Ok(KmsBackend {
             timeout_duration: Duration::from_secs(10),
-            state: Mutex::new(None),
+            state: tokio::sync::Mutex::new(None),
             runtime,
             kms_provider,
         })
     }
 
     pub fn encrypt_content(&self, plaintext: &[u8], iv: Iv) -> Result<EncryptedContent> {
-        let mut opt_state = self.state.lock().unwrap();
+        let runtime = self.runtime.lock().unwrap();
```

Comment:
I see, yeah I guess it could. The problem is that the caller for this method is async and it doesn't allow calling runtime.block in an async context. Let me see if I can move it around to be block less

Reply:
looking at the code again, couldn't find an easy way to make it block less. 

The good thing is that both encrypt/decrypt needs to hold state lock anyway so they are sort of blocking each other in a greater scope already in the original version.

let me know your thoughts!